js/data/bucket/symbol_bucket.js

```diff
@@ -257,12 +258,23 @@ class SymbolBucket extends Bucket {
         const spacing = layout['text-letter-spacing'] * oneEm;
         const textOffset = [layout['text-offset'][0] * oneEm, layout['text-offset'][1] * oneEm];
         const fontstack = this.fontstack = layout['text-font'].join(',');
+        const textAlongLine = layout['text-rotation-alignment'] === 'map' && layout['symbol-placement'] === 'line';
 
         for (const feature of this.features) {
-            let shapedText;
+
+            let shapedTextOrientations;
             if (feature.text) {
-                shapedText = shapeText(feature.text, stacks[fontstack], maxWidth,
-                        lineHeight, horizontalAlign, verticalAlign, justify, spacing, textOffset);
+                const requiresVerticalWritingMode = (
+                    feature.text.match(/[\u1100-\u11FF\uAC00-\uD7A3\u3131-\u318E\u4E00-\u9FCC\u3400-\u4DB5\u3000-\u303F\uFF01-\uFFEE\u3041-\u309F\u30A0-\u30FF\u31F0-\u31FF\uA000-\uA4C6\u1800-\u18AA]/) ||
+                    feature.text.match(/(\uD840[\uDC00-\uFFFF])|[\uD841-\uD872]|(\uD873[\u0000-\uDEAF])/)
```

Comment:
@mourner, per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions:

> Regular expression literals provide compilation of the regular expression when the script is loaded. When the regular expression will remain constant, use this for better performance.

So even if we were to keep this regular expression literal here, it would only be compiled once.